pkg/ebpf/dns_parser.go

```diff
@@ -0,0 +1,107 @@
+package ebpf
+
+import (
+	"bytes"
+
+	"github.com/DataDog/datadog-agent/pkg/process/util"
+	"github.com/google/gopacket"
+	"github.com/google/gopacket/layers"
+)
+
+const maxIPBufferSize = 200
+
+type dnsParser struct {
+	decoder *gopacket.DecodingLayerParser
+	layers  []gopacket.LayerType
+	payload *layers.DNS
+
+	// Cached translation object to reduce allocations
+	cached *translation
+}
+
+func newDNSParser() *dnsParser {
+	payload := &layers.DNS{}
+
+	stack := []gopacket.DecodingLayer{
+		&layers.Ethernet{},
+		&layers.IPv4{},
+		&layers.IPv6{},
+		&layers.UDP{},
+		payload,
+	}
+
+	return &dnsParser{
+		decoder: gopacket.NewDecodingLayerParser(layers.LayerTypeEthernet, stack...),
+		payload: payload,
+		cached:  new(translation),
+	}
+}
+
+func (p *dnsParser) Parse(data []byte) *translation {
+	err := p.decoder.DecodeLayers(data, &p.layers)
+	if err != nil || p.decoder.Truncated {
```

Comment:
nit: I'm curious of whether we care about the error here or not?

Reply:
I *think* for now we have enough telemetry, and I think adding a log entry here could be too spammy.